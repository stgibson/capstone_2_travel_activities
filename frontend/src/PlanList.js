import React from "react";
import { Link } from "react-router-dom";
import { useSelector } from "react-redux";

/**
 * Component for displaying list of currently logged in user's travel plans
 * @returns JSX code for displaying travel plans list
 */
const PlanList = () => {
  const plans = useSelector(store => store.plans);

  return (
    <>
      {
        Object.keys(plans).map(id => (
          <div id={ id } key={ id }>
            <Link to={ `/plans/${id}` }>{ plans[id].name }</Link>
          </div>
        ))
      }
    </>
  );
};

export default PlanList;