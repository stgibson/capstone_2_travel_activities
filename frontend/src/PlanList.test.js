import React from "react";
import { MemoryRouter } from "react-router-dom";
import { createStore } from "redux";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import PlanList from "./PlanList";
import rootReducer from "./reducers/root";

const store = createStore(rootReducer);

it("renders without crashing", () => {
  render(
    <Provider store={ store }>
      <MemoryRouter><PlanList /></MemoryRouter>
    </Provider>
  );
});

it("matches snapshot", () => {
  const { asFragment } = render(
    <Provider store={ store }>
      <MemoryRouter><PlanList /></MemoryRouter>
    </Provider>
  );

  expect(asFragment()).toMatchSnapshot();
});
