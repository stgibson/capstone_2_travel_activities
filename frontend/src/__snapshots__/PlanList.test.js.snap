// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <h2>
    My Travel Plans
  </h2>
</DocumentFragment>
`;
